subprojects/resources-http/src/main/java/org/gradle/internal/resource/transport/http/HttpClientConfigurer.java

```diff
@@ -76,6 +77,8 @@
 
 public class HttpClientConfigurer {
     private static final Logger LOGGER = LoggerFactory.getLogger(HttpClientConfigurer.class);
+    private final int connectionTimeout = 10000;
```

Comment:
I thought about this for a bit and opted for introducing system properties that makes the values configurable. That'll allow us to optimize the timeout in tests.

Additionally, I think it's something users might need. We picked sensible defaults but can't really anticipate user environments. Maybe one of the timeouts is set to low and now the user is completely blocked (thinking about users in China for example). We don't need to document it right now. But if users run into issues then we still have a back door.